test/python/transpiler/test_two_qubit_peephole.py

```diff
@@ -0,0 +1,439 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2025
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=missing-function-docstring
+
+"""
+Tests for the default UnitarySynthesis transpiler pass.
+"""
+
+import math
+import numpy as np
+import ddt
+
+from qiskit import generate_preset_pass_manager
+from qiskit.providers.fake_provider import GenericBackendV2
+from qiskit.circuit import QuantumCircuit, QuantumRegister
+from qiskit.circuit.parameterexpression import ParameterValueType
+from qiskit.converters import circuit_to_dag, dag_to_circuit
+from qiskit.quantum_info.operators import Operator
+from qiskit.quantum_info.random import random_unitary
+from qiskit.transpiler import PassManager, CouplingMap, Target, InstructionProperties
+from qiskit.transpiler.passes import TwoQubitPeepholeOptimization, TrivialLayout
+from qiskit.circuit.library import (
+    IGate,
+    CXGate,
+    RZGate,
+    RXGate,
+    SXGate,
+    XGate,
+    iSwapGate,
+    ECRGate,
+    UGate,
+    ZGate,
+    RYYGate,
+    RZZGate,
```

Comment:
According to the recent [error log](https://dev.azure.com/qiskit-ci/qiskit/_build/results?buildId=66550&view=logs&j=ee6a574e-a884-556d-e11f-f4e0408370dd&t=cda6e954-50e0-5d66-9385-47256a656c01), RZZGate is not used.
```suggestion
```